### PR TITLE
cli to view snark pool

### PR DESCRIPTION
**Pull request by deepthiskumar**
client command `coda advanced snark-pool-list` to view all snark works in the snark pool

```
[{"work_ids":[388496774],"fee":1,"prover":"8QnLYnzAkhx3moXHHBBpK418LGCEiQZzYDgKggntsNkZqsUJLFos474Mzoy5mWJP9S"},{"work_ids":[298576701],"fee":1,"prover":"8QnLYnzAkhx3moXHHBBpK418LGCEiQZzYDgKggntsNkZqsUJLFos474Mzoy5mWJP9S"},{"work_ids":[388496774,869707141],"fee":1,"prover":"8QnLYnzAkhx3moXHHBBpK418LGCEiQZzYDgKggntsNkZqsUJLFos474Mzoy5mWJP9S"},{"work_ids":[298576701,520748931],"fee":1,"prover":"8QnLYnzAkhx3moXHHBBpK418LGCEiQZzYDgKggntsNkZqsUJLFos474Mzoy5mWJP9S"}]
```

Checklist:

- [ ] Tests were added for the new behavior
- [ ] All tests pass (CI will check this if you didn't)
- [ ] Serialized types are in stable-versioned modules
- [ ] Does this close issues? List them:

### CODE REVIEW

File: src/lib/daemon_rpcs/daemon_rpcs.ml [hunk 1]
```diff
@@ -512,6 +512,17 @@ module Snark_job_list = struct
     Rpc.Rpc.create ~name:"Snark_job_list" ~version:0 ~bin_query ~bin_response
 end
 
+module Snark_pool_list = struct
```
Comment (wu-s-john) [inline src/lib/daemon_rpcs/daemon_rpcs.ml]: We are not trying to create RPCs anymore. Can you write `Snark_pool_list` as a graphql service?

Comment (deepthiskumar) [reply src/lib/daemon_rpcs/daemon_rpcs.ml]: Why don't we want to create RPCs anymore? Also, this is mainly for next week's snark challenge.

**Comment by deepthiskumar** [comment]
> Make sure, you convert this to a GraphQL query

I'll make an issue

**Comment by jkrauska** [comment]
> John is right, but since we really need this in for this release I’ll let it in — please move to graphql later today though

please /add/ also to graphql? 

cli still matters until the cli uses graphql.

**Comment by Schmavery** [comment]
@jkrauska to clarify, I think john was saying to replace the rpc functionality with graphql, not replace the cli with graphql -- we can use graphql endpoints from cli commands now.

File: src/app/cli/src/init/client.ml [hunk 2]
```diff
@@ -683,6 +683,17 @@ let snark_job_list =
...
              print_rpc_error e ))
 
+let snark_pool_list =
+  let open Deferred.Let_syntax in
+  let open Command.Param in
+  Command.async ~summary:"List of snark works in the snark pool in JSON format"
+    (Cli_lib.Background_daemon.init (return ()) ~f:(fun port () ->
+         match%map dispatch Daemon_rpcs.Snark_pool_list.rpc () port with
+         | Ok str ->
+             printf "%s" str
```
Comment (cmr) [inline src/app/cli/src/init/client.ml]: I cringe every time I see a `printf "%s" s` as opposed to a `Stdout.write s` or similar.

File: src/lib/daemon_rpcs/daemon_rpcs.ml [hunk 3]
```diff
@@ -516,6 +516,17 @@ module Snark_job_list = struct
...
+  type query = unit [@@deriving bin_io]
+
+  type response = string [@@deriving bin_io]
```
Comment (cmr) [inline src/lib/daemon_rpcs/daemon_rpcs.ml]: I would have preferred exposing a `Yojson.Safe.t` (which probably doesn't have bin_prot stuff...), but a string here seems fine for the intended use.

**Pull request merged**